workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/androidx/WorkflowLifecycleOwner.kt

```diff
@@ -190,66 +192,44 @@ internal class RealWorkflowLifecycleOwner(
       oldLifecycle?.removeObserver(this)
       parentLifecycle?.addObserver(this)
     }
-    updateLifecycle()
+    updateLifecycle(Event.ON_START)
   }
 
   override fun onViewDetachedFromWindow(v: View) {
     viewIsAttachedToWindow = false
-    updateLifecycle()
+    updateLifecycle(ON_PAUSE)
   }
 
   /** Called when the [parentLifecycle] changes state. */
   override fun onStateChanged(
     source: LifecycleOwner,
     event: Event
   ) {
-    updateLifecycle()
+    // we should always consume what the new event is from onStateChanged. if we try to
+    // assume the event like we did in the past, we can get into scenarios where we try to put
+    // a view in created, when it's parent hasn't completed fully it's super.onCreate yet.
+    updateLifecycle(event)
   }
 
   override fun destroyOnDetach() {
     if (!destroyOnDetach) {
       destroyOnDetach = true
-      updateLifecycle()
+      updateLifecycle(Event.ON_DESTROY)
```

Comment:
This subverts the purpose of `destroyOnDetach()`, which is to make sure things are still alive while the view is visible. I think instead you want to change `onViewDetachedFromWindow` to look at `destroyOnDetach` and use that to choose between `ON_PAUSE` and `ON_DESTROY`. Maybe also need to change the guard clause at the top of `onViewAttachedToWindow` to look directly at `destroyOnDetach`.